test/retry_policy_tests.py

```diff
@@ -221,7 +221,7 @@ def test_default_retry_policy_for_query(self):
         result_docs = list(docs)
         self.assertEqual(result_docs[0]['id'], 'doc1')
         self.assertEqual(result_docs[1]['id'], 'doc2')
-        self.assertEqual(self.counter, 12)
+        self.assertEqual(self.counter, 6)
```

Comment:
This test counts how many retries were done.  Before the fix, 12 http requests were generated.  Now 6 are.  This is by-design.  Updating test to match.